packages/synapse-react-client/src/components/ColumnFilter/ColumnFilter.test.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import { QueryResultBundle } from '@sage-bionetworks/synapse-types'
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import ColumnFilter from './ColumnFilter'
+import syn20337467Json from '../../../src/mocks/query/syn20337467.json'
+import { useQuery } from '@tanstack/react-query'
+import { useQueryContext } from '../QueryContext/QueryContext'
+
+jest.mock('../QueryContext/QueryContext', () => ({
+  useQueryContext: jest.fn(),
+}))
+jest.mock('@tanstack/react-query', () => ({
+  useQuery: jest.fn(),
+  QueryClient: jest.fn().mockImplementation(() => ({
+    queryCache: { clear: jest.fn() },
+    setQueryData: jest.fn(),
+    getQueryData: jest.fn(),
+  })),
+}))
+
+const data = syn20337467Json as QueryResultBundle
+
+describe('ColumnFilter tests', () => {
+  const executeQueryRequest = jest.fn()
+  const addValueToSelectedFacet = jest.fn()
+  const removeSelectedFacet = jest.fn()
+
+  beforeEach(() => {
+    ;(useQueryContext as jest.Mock).mockReturnValue({
```

Comment:
You can get type-safe mocks with the following pattern:

const mockUseQueryContext = jest.mocked(useQueryContext)

```suggestion


describe('ColumnFilter tests', () => {
  const executeQueryRequest = jest.fn()
  const addValueToSelectedFacet = jest.fn()
  const removeSelectedFacet = jest.fn()

  beforeEach(() => {
   mockUseQueryContext.mockReturnValue({
```

Please remove all of these unnecessary casts where you could use `jest.mocked`